src/views/SignTransactionLedger.vue

```diff
@@ -148,8 +153,19 @@ export default class SignTransactionLedger extends Vue {
 
             senderUserFriendlyAddress = this.$store.state.activeUserFriendlyAddress;
 
+            /*
+             * There has to be a NIM paymentOption so find will always return a result.
+             */
+            const recipient = (this.request as ParsedSignTransactionRequest).recipient
+                || ((this.request as ParsedCheckoutRequest).paymentOptions.find(
+                        (option) => option.currency === Currency.NIM,
+                    ) as ParsedNimiqDirectPaymentOptions).protocolSpecific.recipient;
```

Comment:
I feel this finding of the Nimiq PaymentOption is so much used, it warrants its own helper function. This helper would also be typed correctly, so no need for this repetition everywhere.